[Minhao23]

minhao23 opened the pull request:
# Jarvis

> “throughout heaven and earth, I alone am the cooked one" 

Jarvis frees your mind from having to remember things you need to do. It's:

- **Text-based**
- **Easy to learn**
- **FAST** – *SUPER FAST* to use

All you need to do is:

1. **Download** it from [here](#).
2. **Double-click** it.
3. **Add** your tasks.
4. **Let it manage your tasks for you** 😉.

And the best part? It’s **FREE**!

## Features

- [x] Managing tasks
- [ ] Managing deadlines (*coming soon*)
- [ ] Reminders (*coming soon*)

## To run Jarvis
```
Jarvis Jarvis = new Jarvis();
jarvis.main();
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

minhao23 committed:
test

minhao23 committed:
Level 0. Rename, Greet, Exit

minhao23 committed:
Level 1. Echo

minhao23 committed:
Level 2. Add, List

minhao23 committed:
Level 3. Mark as Done

minhao23 committed:
Level 4. ToDos, Events, Deadlines

minhao23 committed:
Level-5: Handle Errors

minhao23 committed:
Level 6. Delete

minhao23 committed:
Level 7. Save

minhao23 committed:
Level 7. Save

minhao23 committed:
Merge branch 'add-gradle-support'

minhao23 committed:
add test files

minhao23 committed:
Level 8

minhao23 committed:
Add Gradle to Level 8

minhao23 committed:
add branch level 9

minhao23 committed:
add javadoc

minhao23 committed:
Merge branch 'javadoc'

# Conflicts:
#	ip/src/main/java/jarvis/Parser.java

minhao23 committed:
add checkstyle

minhao23 committed:
Add back the handleDelete() method

minhao23 committed:
test commit for repo sense

minhao23 committed:
Change file directory, remove redundant ip directory

minhao23 committed:
reorganise file directory, debug test

minhao23 committed:
reorganise file directory, debug test

minhao23 committed:
add gui for chatbot

minhao23 committed:
add gui for chatbot

minhao23 committed:
add level 10 tag

minhao23 committed:
reorganise file directory, debug test

minhao23 committed:
reorganise file directory, debug test

minhao23 committed:
reorganise file directory, debug test

minhao23 committed:
Remove .idea/ from version control

minhao23 committed:
t#	deleted:    ip/.idea/caches/7.6.2/scripts/9fump2pqpf0xi0z6v987wkigy/init/GradleDaemonClasspathEnhancer.class

minhao23 committed:
test

minhao23 committed:
fix minor bug regarding parsing

JunLongling commented on src/main/java/jarvis/ui/MainWindow.java:
```diff
@@ -0,0 +1,65 @@
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        String logo =
+                "       _                  _     \n" +
+                        "      (_)___ _______   __(_)____\n" +
+                        "     / / __ `/ ___/ | / / / ___/\n" +
+                        "    / / /_/ / /   | |/ / (__  )\n" +
+                        " __/ /\\__,_/_/    |___/_/____/\n" +
```
it might be a good idea to move the + operator to a new line when concatenating strings. This could help make the code easier to read and maintain. What do you think?

JunLongling commented on src/main/java/jarvis/logic/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+            }
+            default -> {
+                throw new IllegalArgumentException();
+            }
```
Great use of the modern switch syntax! Just a small suggestion: the case clauses are currently indented, but standard Java style recommends that they should be aligned directly under the switch statement without indentation. You might want to adjust your IDE settings to follow [this style](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=The%20switch%20statement%20should%20have%20the%20following%20form%3A%20Note%20there%20is%20no%20indentation%20for%20case%20clauses) for consistency and readability. What do you think?

JunLongling commented on src/main/java/jarvis/logic/Storage.java:
```diff
@@ -0,0 +1,177 @@
+package jarvis.logic;
+
+import java.io.*;
```
I noticed you're using a wildcard import (import java.io.*). It's generally a good practice to list imported classes explicitly rather than using wildcards, as it improves clarity and avoids potential naming conflicts. Would you consider adjusting it?

JunLongling reviewed (commented):
Great job on adhering to coding standards—everything looks good from my perspective! 🎉 To help catch any potential style violations and ensure consistency, I would recommend using [CheckStyle](https://se-education.org/guides/tutorials/checkstyle.html) to detect coding style issues automatically. It's a great tool to further enhance the quality of your code. Keep up the awesome work! 👍

fymbc commented on src/main/java/jarvis/logic/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+
+    LocalDate dateDeadline;
+    String deadline;
+    public Deadline(String name, String deadline){
```
Perhaps leave a whitespace before '{' ? I noticed the same issue in a few other places too!

fymbc commented on src/main/java/jarvis/logic/Event.java:
```diff
@@ -0,0 +1,42 @@
+
+public class Event extends Task {
+
+    String start;
+    String end;
+    LocalDate dateStart;
+    LocalDate dateEnd;
```
While not a coding standard violation, perhaps it would be best to put a `private `access modifier (and/or even `final`) before the fields? I noticed the same issue in a few other places too!

fymbc commented on src/main/java/jarvis/logic/Jarvis.java:
```diff
@@ -0,0 +1,59 @@
+            this.parser = new Parser();
+            this.storage = Storage.getInstance();
+        }
+
```
I believe this is one level indentation too many, maybe shift the indentation to the left?

fymbc commented on src/main/java/jarvis/logic/Parser.java:
```diff
@@ -0,0 +1,109 @@
+                    result.append("bad input: invalid format ").append(input).append("\n");
+                } catch (IllegalArgumentException e) {
+                    result.append("Invalid input.\n");
+                } catch (Exception e) {
```
I'm not too sure if I'm correct on this, but I think Pokemon exception handling here wouldn't be ideal as a catch-all statement because we wouldn't know where went wrong if there is a new exception expect. What do you think?

fymbc commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,9 @@
-| | | | | | | |/ / _ \
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
+       _                  _
+      (_)___ _______   __(_)____
+     / / __ `/ ___/ | / / / ___/
+    / / /_/ / /   | |/ / (__  )
+ __/ /\__,_/_/    |___/_/____/
+/___/
```
Nice touch! I really like this :)

fymbc reviewed (commented):
LGTM! I believe just some indentations and whitespaces to fix.

To catch these issues, I suggest trying using the Gradle checkstyleMain functionality or Checkstyle-IDEA plugin to catch things that we may have missed out. 👍

minhao23 committed:
Add the functionality for "bye" and storage

minhao23 committed:
Fix the previous bugs in Jarvis

JarvisDialog is now printing empty Strings, this is stylistically not user friendly.
HandleDelete has an off by 1 error, always returning an extra item in the list.

Let's
* update HandleDelete to remove the off by one error
* update JarvisDialog to not print empty Strings

minhao23 committed:
Restructure code quality

There are several methods that currently have deep nesting, which is poor for readability.

Refactor various methods to comply with the coding standard.

minhao23 committed:
Merge pull request #1 from minhao23/branch-A-CodeQuality

Restructure code quality

minhao23 committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/jarvis/logic/TaskList.java
#	src/main/java/jarvis/ui/MainWindow.java

minhao23 committed:
Merge pull request #2 from minhao23/branch-A-Assertions

Fix the previous bugs in Jarvis

minhao23 committed:
Create gradle.yml

minhao23 committed:
Improve UI display of Dialogbox

minhao23 committed:
Add user guide

The current user guide is the default one based on Duke.

In order to fix this, let's change to one based on Jarvis

minhao23 committed:
Tidy up file directory

minhao23 committed:
Update storage file to absolute address

minhao23 committed:
Update storage file to absolute address

minhao23 committed:
Add more personality features, minor bug fixes

minhao23 committed:
Fix the resizing issue

minhao23 committed:
Fix minor output String bugs

minhao23 committed:
Add reformatting option for Dates

minhao23 committed:
Fix minor issues with marking logic
